pkg/config/validation/validation.go

```diff
@@ -1106,10 +1114,11 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 					}
 					udsMap[bind] = struct{}{}
 				} else {
-					if _, found := portMap[egress.Port.Number]; found {
+					p := ipPortPair{ip: egress.Bind, port: egress.Port.Number}
+					if _, found := portMap[p]; found {
```

Comment:
similarly to the `ingress` listener case, I suggest to add unit tests for the config generator on newly supported scenarios